public/app/core/logs_model.ts

```diff
@@ -326,3 +324,174 @@ export function makeSeriesForLogs(rows: LogRowModel[], intervalMs: number): Time
     };
   });
 }
+
+export function seriesDataToLogsModel(seriesData: SeriesData[], intervalMs: number): LogsModel {
+  const metricSeries: SeriesData[] = [];
+  const logSeries: SeriesData[] = [];
+
+  for (const series of seriesData) {
+    const logSeriesFieldProcessor = new LogsSeriesFieldProcessor(series);
```

Comment:
Maybe we could new up this once and then process the series in the loop?

Reply:
You mean so that it can be used in line 382 as well, i.e. use an array of processors instead of looping seriesData?

Reply:
Well I was thinking of new:ing up objects in a loop, don't know how many objects there are in this case. As the rest of the discussion seemed to be leaned towards performance I added this. No biggie.

Reply:
Maximum 28 or something I suppose - the maximum of `refId`, a b c etc.

Reply:
This is now gone :boom: